File: classroom/src/types/classes.ts
import { DateTime } from "luxon";

export type ClassesData = {
  /** 授業があるか */
  isEmpty: boolean;

  /** 授業リスト */
  items: ClassData[];
};

export type ClassData = {
  /** 授業名 */
  title: string;
  /** 何時間目か */
  period: number;

  /** 開始時間 */
  startAt: DateTime;
  /** 終了時間 */
  endAt: DateTime;
};

File: classroom_flutter/lib/models/classes.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'classes.freezed.dart';
part 'classes.g.dart';

@freezed
class ClassList with _$ClassList {
  const factory ClassList({
    required List<ClassData> items,
  }) = _ClassList;

  factory ClassList.fromJson(Map<String, dynamic> json) =>
      _$ClassListFromJson(json);
}

@freezed
class ClassData with _$ClassData {
  const factory ClassData({
    required String title,
    required int period,
    required DateTime startAt,
    required DateTime endAt,
  }) = _ClassData;

  factory ClassData.fromJson(Map<String, dynamic> json) =>
      _$ClassDataFromJson(json);
}
